packages/SwingSet/tools/bundleTool.js

```diff
@@ -6,6 +6,9 @@ import styles from 'ansi-styles'; // less authority than 'chalk'
 
 const { quote: q, Fail } = assert;
 
+/** @type {Map<string, Promise<BundleMeta>>} output filename to its meta info */
+const fileMetas = new Map();
```

Comment:
does this state need to be shared across bundle caches?

Reply:
yes because each multiple bundle caches are trying to write to the same file paths. I suppose it would be better for "makeBundleCache" to be idempotent. I'll work on that. At least now for this PR I know that the failure is in the bundleTool

Reply:
@michaelfig you recently did some work to address concurrent writes; take a look at this, please?

see also #3609

Reply:
> yes because each multiple bundle caches are trying to write to the same file paths.

I wonder if they're in the same process / heap, though. What @michaelfig did was to add .lock files, IIRC.

Reply:
@turadg, can you document the issue you ran into with the bundle cache implementation that existed before this PR?  I'm most curious how the problem manifested, such as unexpected output or exceptions.

The `.lock` file handshake should have been idempotent and robust against multiple uses of the same cache, whether or not they are inside the same process or multiple processes (including `bundleTool.js`).

Reply:
It's visible in the previous build: https://github.com/Agoric/agoric-sdk/commit/7675e71bf6d9b98ae12b4bd7e00e7700e584be95

It [failed](https://github.com/Agoric/agoric-sdk/actions/runs/4846575703/jobs/8636244952) but only in CI,
```
#   create initial version
#   now perform the upgrade
[bundleTool] bundles add: vat-board from /home/runner/work/agoric-sdk/agoric-sdk/packages/vats/src/vat-board.js
[bundleTool] bundles add: vat-bridge from /home/runner/work/agoric-sdk/agoric-sdk/packages/vats/src/vat-bridge.js
[bundleTool] bundles add: vat-priceAuthority from /home/runner/work/agoric-sdk/agoric-sdk/packages/vats/src/vat-priceAuthority.js
[bundleTool] bundles add: bootstrap-relay from /home/runner/work/agoric-sdk/agoric-sdk/packages/SwingSet/test/bootstrap-relay.js
[bundleTool] bundles add: bootstrap-relay from /home/runner/work/agoric-sdk/agoric-sdk/packages/SwingSet/test/bootstrap-relay.js
[bundleTool] bundles add: bootstrap-relay from /home/runner/work/agoric-sdk/agoric-sdk/packages/SwingSet/test/bootstrap-relay.js
REJECTED from ava test("upgrade vat-board"): (Error#1)
Error#1: config.vats.bootstrap: Invalid or unexpected token
  at packages/SwingSet/src/controller/initializeSwingset.js:538:15
  at async Promise.all (index 0)
  at async processGroup (packages/SwingSet/src/controller/initializeSwingset.js:541:27)
  at async initializeSwingset (packages/SwingSet/src/controller/initializeSwingset.js:571:31)
  at async buildVatController (packages/SwingSet/src/controller/controller.js:479:23)
  at async makeScenario (packages/vats/test/upgrading/test-upgrade-vats.js:53:13)
  at async packages/vats/test/upgrading/test-upgrade-vats.js:74:18
```

Not it bundles `bootstrap-relay` once because the cache is shared by the three tests.

Reply:
> It's visible in the previous build: ...

Thanks for sharing that.

Reply:
Oops.  Turns out in the bustle of trying to sync Endo, I forgot to create an Agoric SDK PR to have `bundleTool.js` use Endo's idempotent-between-processes bundle cache.

Now I did that in #7596.